include/lapack/lu_mult.hpp

```diff
@@ -0,0 +1,105 @@
+/// @file lu_mult.hpp
+/// @author Lindsay Slager, University of Colorado Denver, USA
+//
+// Copyright (c) 2013-2022, University of Colorado Denver. All rights reserved.
+//
+// This file is part of <T>LAPACK.
+// <T>LAPACK is free software: you can redistribute it and/or modify it under
+// the terms of the BSD 3-Clause license. See the accompanying LICENSE file.
+
+#ifndef __TLAPACK_LU_MULT_HH__
+#define __TLAPACK_LU_MULT_HH__
+
+#include "base/utils.hpp"
+#include "base/types.hpp"
+
+namespace tlapack {
+
+template <typename idx_t>
+struct lu_mult_opts_t
+{
+    // Optimization parameter. Matrices smaller than nx will not
+    // be multiplied using recursion. Must be at least 1.
+    idx_t nx = 1;
+};
+
+/**
+ *
+ * @brief in-place multiplication of lower triangular matrix L and upper triangular matrix U.
+ *      this is the recursive variant
+ * 
+ * @param[in,out] A n-by-n matrix
+ *      On entry, the strictly lower triangular entries of A contain the matrix L.
+ *      L is assumed to have unit diagonal.  
+ *      The upper triangular entires of A contain the matrix U. 
+ *      On exit, A contains the product L*U. 
+ * @param[in] (Optional)  struct containing optimization parameters. See lu_mult_ops_t
+ * 
+ * @ingroup util
+ */
+template <class matrix_t>
+void lu_mult(matrix_t &A, const lu_mult_opts_t<size_type<matrix_t>> &opts = {})
+{
+    using idx_t = size_type< matrix_t >;;
+    using T = type_t<matrix_t>;
+    using range = std::pair<idx_t, idx_t>;
+    using real_t = real_type<T>;
+
+    const idx_t m = nrows(A);
+    const idx_t n = ncols(A);
+
```

Comment:
You should assert that `nrows(A) == ncols(A)` and than `opts.nx >= 1`

Reply:
Solved!